resilience4j-core/src/main/java/io/github/resilience4j/core/metrics/Snapshot.java

```diff
@@ -0,0 +1,87 @@
+/*
+ *
+ *  Copyright 2019 Robert Winkler and Bohdan Storozhuk
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *
+ */
+package io.github.resilience4j.core.metrics;
+
+import java.time.Duration;
+
```

Comment:
Thinking loud here , Small question for discussion , can we maybe have Snapshot<T> and same for Metrics<T> and let the user define what type of APIs/counts he can use for metrics by call get T . (api call) or we should enforce what type of stats he can get and should follow that ?

Reply:
Can you give an example? I don't understand it

Reply:
per discussion not applicable